flow360/component/simulation/models/volume_models.py

```diff
@@ -0,0 +1,152 @@
+
+
+class NavierStokesModifiedRestartSolution(NavierStokesInitialCondition):
+    type: Literal["restartManipulation"] = pd.Field("restartManipulation", frozen=True)
+
+
+class HeatEquationInitialCondition(ExpressionInitialConditionBase):
+    temperature: str = pd.Field(displayed="T [non-dim]")
+
+
+class PDEModelBase(Flow360BaseModel):
+    """
+    Base class for equation models
+
+    """
+
+    material: MaterialType = pd.Field()
+    initial_condition: Optional[dict] = pd.Field(None)
+
+
+class FluidDynamics(PDEModelBase):
+    """
+    General FluidDynamics volume model that contains all the common fields every fluid dynamics zone should have.
+    """
+
+    navier_stokes_solver: NavierStokesSolver = pd.Field(NavierStokesSolver())
+    turbulence_model_solver: TurbulenceModelSolverType = pd.Field(SpalartAllmaras())
+    transition_model_solver: Optional[TransitionModelSolver] = pd.Field(None)
+
+    material: MaterialTypes = pd.Field(Air())
+
+    initial_condition: Optional[
+        Union[NavierStokesModifiedRestartSolution, NavierStokesInitialCondition]
+    ] = pd.Field(None)
+
+
+class HeatTransfer(PDEModelBase):
+    """
+    General HeatTransfer volume model that contains all the common fields every heat transfer zone should have.
+    """
+
+    entities: EntityList[GenericVolume, str] = pd.Field(alias="volumes")
+
+    heat_equation_solver: HeatEquationSolver = pd.Field(HeatEquationSolver())
+    volumetric_heat_source: Union[HeatSourceType, pd.StrictStr] = pd.Field(0)
+
+    initial_condition: Optional[HeatEquationInitialCondition] = pd.Field(None)
+
+
+class ActuatorDisk(Flow360BaseModel):
+    """Same as Flow360Param ActuatorDisks.
+    Note that `center`, `axis_thrust`, `thickness` can be acquired from `entity` so they are not required anymore.
+    """
+
+    entities: Optional[EntityList[Cylinder]] = pd.Field(None, alias="volumes")
+
+    force_per_area: ForcePerArea = pd.Field()
+
+
+class BETDisk(Flow360BaseModel):
+    """Same as Flow360Param BETDisk.
+    Note that `center_of_rotation`, `axis_of_rotation`, `radius`, `thickness` can be acquired from `entity` so they are not required anymore.
+    """
+
+    entities: Optional[EntityList[Cylinder]] = pd.Field(None, alias="volumes")
+
+    rotation_direction_rule: Literal["leftHand", "rightHand"] = pd.Field("rightHand")
+    radius: Optional[LengthType.Positive] = pd.Field(None)
+    number_of_blades: pd.conint(strict=True, gt=0, le=10) = pd.Field()
+    omega: AngularVelocityType.NonNegative = pd.Field()
+    chord_ref: LengthType.Positive = pd.Field()
+    n_loading_nodes: pd.conint(strict=True, gt=0, le=1000) = pd.Field()
+    blade_line_chord: LengthType.NonNegative = pd.Field(0)
+    initial_blade_direction: Optional[Axis] = pd.Field(None)
+    tip_gap: Union[LengthType.NonNegative, Literal["inf"]] = pd.Field("inf")
+    mach_numbers: List[NonNegativeFloat] = pd.Field()
+    reynolds_numbers: List[PositiveFloat] = pd.Field()
+    alphas: List[float] = pd.Field()
```

Comment:
From Maciej: "For alpha, beta, can we just assume these are in degrees and take floats, not unit quantities? This is will be so easy to make mistake because default units of angle are radians. Alternatively, we can introduce another unit in the unit system but this seems to be an unnecessary complication"